pkg/generator/ingress_translator.go

```diff
@@ -192,13 +196,17 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 
 				var transportSocket *envoycorev3.TransportSocket
 
-				// This has to be "OrDefaults" because this path is called before the informers are
+				// This has to be "OrDefaults" because this path could be called before the informers are
```

Comment:
Could we just avoid setting up the envoy configuration too early eg. when creating a new controller? For example right now we expect each controller to set is own cm watches, call Watch on the passed cmw for each cmw and then later on we Start() the cmw.
I propose to solve this once and for all and do either:
a) allow a controller to have a postCreate func to be called after cmw is started (if that is useful as in Kourier).
b) Allow starting cmw in the constructor and skip it at shared main.
The only thing you need to be careful is not to start it too early as you will hit issues like [this](https://github.com/knative/eventing/pull/6615/files#diff-2a10da0d0184623583740c11a53b98f7cb293507b027dfa19622e02e08ef4e10).

Reply:
Avoiding to set up the envoy configuration is also fine for me. But I am not so familiar with controller stuff so I would like to ask you to send the PR.

Reply:
The easiest way would be to fetch the cm imho. Let's see what @dprotaso has to say. If we want to add flexibility at sharedmain I can send a PR for sure.

Reply:
@nak3 check here: https://github.com/knative/serving/blob/3812acfe03bce45acd332c7706ff1564446425cb/cmd/autoscaler/main.go#L129